src/commands/addMIConnections/ConnectionsListStep.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type StringDictionary } from "@azure/arm-appservice";
+import { isSettingConvertible } from "@microsoft/vscode-azext-azureappsettings";
+import { AzExtFsExtra, AzureWizardPromptStep, nonNullValue, type IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { type ILocalSettingsJson } from "../../funcConfig/local.settings";
+import { localize } from "../../localize";
+import { decryptLocalSettings } from "../appSettings/localSettings/decryptLocalSettings";
+import { encryptLocalSettings } from "../appSettings/localSettings/encryptLocalSettings";
+import { getLocalSettingsFile } from "../appSettings/localSettings/getLocalSettingsFile";
+import { type AddMIConnectionsContext } from "./AddMIConnectionsContext";
+
+export interface Connection {
+    name: string;
+    value: string;
+}
+
+export class ConnectionsListStep extends AzureWizardPromptStep<AddMIConnectionsContext> {
+    public async prompt(context: AddMIConnectionsContext): Promise<void> {
+        const picks = await this.getPicks(context);
+
+        if (picks.length === 0) {
+            const noItemFoundMessage: string = localize('noConnectionsFound', 'No connections found in local settings');
+            (await context.ui.showQuickPick(picks, {
+                placeHolder: localize('selectConnections', 'Select the connections you want to add managed identity support for'),
+                suppressPersistence: true,
+                noPicksMessage: noItemFoundMessage
+            }));
+        } else {
+            context.connections = (await context.ui.showQuickPick(picks, {
```

Comment:
Do you really need to have two separate calls for this? If the picks array is empty, isn't it going to have the same behavior as it does in the if clause?

Reply:
Talked in person but we used two seperate calls since one is using `canPickMany` and if there are no picks we don't want `canPickMany` to be true.